packages/dds/tree/src/simple-tree/api/schemaFactory.ts

```diff
@@ -978,6 +981,25 @@ export class SchemaFactory<
 		return out;
 	}
 
+	private getStructuralType(
+		fullName: string,
+		types: TreeNodeSchema | readonly TreeNodeSchema[],
+		builder: () => TreeNodeSchema,
+	): TreeNodeSchema {
+		const structural = getOrCreate(this.structuralTypes, fullName, builder);
+		const inputTypes = new Set(normalizeAllowedTypes(types));
+		const outputTypes = new Set(
+			normalizeAllowedTypes(structural.info as TreeNodeSchema | readonly TreeNodeSchema[]),
+		);
+		const same = compareSets({ a: inputTypes, b: outputTypes });
+		if (!same) {
+			throw new UsageError(
+				`Structurally named schema collision: two schema named ${fullName} were defined with different input schema.`,
```

Comment:
Nit: I generally recommend throwing quotes around names in error messages to help them stand out / prevent situations where the meaning of the message could be inadvertently altered.
```suggestion
				`Structurally named schema collision: two schema named "${fullName}" were defined with different input schema.`,
```